conans/test/functional/profiles/test_python_profiles.py

```diff
@@ -0,0 +1,99 @@
+import os
+import textwrap
+import unittest
+
+from importlib.util import spec_from_file_location, module_from_spec
+
+from conans.client.tools import environment_append
+from conans.test.utils.tools import TestClient
+
+
+class Profile(dict):
+
+    def __init__(self):
+        self["settings"] = {}
+        self["options"] = {}
+        self["env"] = {}
+
+    @property
+    def settings(self):
+        return self["settings"]
+
+    def update(self, profile):
+        for k, v in profile.items():
+            if k in ["settings", "options", "env"]:
+                self[k].update(v)
+            else:
+                self[k] = v
+
+
+def _load_module_from_file(filename, parent_profile):
+    name = os.path.splitext(os.path.basename(filename))[0]
+    spec = spec_from_file_location(name, filename)
+    config = module_from_spec(spec)
+    for key, value in parent_profile.items():
+        setattr(config, key, value)
+    spec.loader.exec_module(config)
+    return config
+
+
+def _load_profile_python(profile_path, parent_profile=None):
+    # TODO: implement include() of other profiles? variable substitution?
+    parent_profile = parent_profile or Profile()
+    profile_content = _load_module_from_file(profile_path, parent_profile)
+    profile = Profile()
+    for key, value in profile_content.__dict__.items():
+        if not key.startswith("__"):
+            profile[key] = value
```

Comment:
here we set all the variables to the profile in order to achieve generic variable substitution